src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -1974,18 +2174,13 @@ class ArnoldLight : public ArnoldObject
 				}
 			}
 
-			// Deal with light filters.
-
-			const std::vector<ArnoldShaderPtr> &lightFilterShaders = arnoldAttributes->lightFilterShaders();
-			AtArray *linkedFilterNodes = AiArrayAllocate( lightFilterShaders.size(), 1, AI_TYPE_NODE );
-			for( unsigned i = 0; i < lightFilterShaders.size(); ++i )
-			{
-				AiArraySetPtr( linkedFilterNodes, i, lightFilterShaders[i]->root() );
-			}
+			// Deal with light filter connections
 
-			AiNodeSetArray( m_lightShader->root(), g_filtersArnoldString, linkedFilterNodes );
+			m_lightFilterShaders = arnoldAttributes->lightFilterShaders();
+			m_connections->registerLight( m_name, this );
```

Comment:
This is re-registering the light every single time a light shader parameter is tweaked. Can't we just register once in the constructor?

Reply:
Only if we add another way to dirty a light, I think? It's a little misleading, but what this is essentially doing is signaling that the internal shader has been updated and that we'll need to set the blockers again. The fact that we throw away the `ArnoldShader` for every change means that we also throw away the blocker connections. Without this call, `updateFilters() `won't get called.

Do you want me to add a separate method that dirties a light? Do you see a better way of doing this? I don't remember why I opted for this way of doing things - maybe just to keep the `BlockerConnections` API small and you _could_ think of this as registering the new light (shader). Maybe it's sufficient to just add a comment in the code that explains it?